### PR TITLE
test: mirgate fix* and future_incompat_report to snapbox

### PR DESCRIPTION
**Pull request by eth3lbert**
### What does this PR try to resolve?

Part of https://github.com/rust-lang/cargo/issues/14039.

Migrate following to snapbox:

- `tests/testsuite/fix.rs`
- `tests/testsuite/fix_n_times.rs`
- `tests/testsuite/future_incompat_report.rs`

### CODE REVIEW

File: tests/testsuite/fix.rs [hunk 1]
```diff
@@ -1342,12 +1385,16 @@ fn fix_in_existing_repo_weird_ignore() {
...
+        .with_stderr_data(str![[r#"
+[ERROR] no VCS found for this package and `cargo fix` can potentially perform destructive changes; if you'd like to suppress this error pass `--allow-no-vcs`
+
+"#]])
         .with_status(101)
         .run();
     p.cargo("fix").cwd("src").run();
 }
 
+#[allow(deprecated)]
```
Comment (eth3lbert) [inline tests/testsuite/fix.rs:1397]: `with_stderr_does_not_contain` is being used in this function.

File: tests/testsuite/fix.rs [hunk 2]
```diff
@@ -1618,16 +1677,20 @@ fn abnormal_exit() {
...
+...
+I'm not a diagnostic.
+rustc exited abnormally: [..]
+Original diagnostics will follow.
+...
+"#]])
         .run();
 }
 
+#[allow(deprecated)]
```
Comment (eth3lbert) [inline tests/testsuite/fix.rs:1693]: `with_stderr_does_not_contain` is being used in this function.

File: tests/testsuite/future_incompat_report.rs [hunk 3]
```diff
@@ -68,25 +74,24 @@ fn test_zero_future_incompat() {
...
-        )
+        .with_stderr_data(str![[r#"
+[FINISHED] `dev` profile [unoptimized + debuginfo] target(s) in [ELAPSED]s
+[NOTE] 0 dependencies had future-incompatible warnings
+
+"#]])
         .run();
 }
 
+#[allow(deprecated)]
```
Comment (eth3lbert) [inline tests/testsuite/future_incompat_report.rs:94]: `with_stderr_does_not_contain` is being used in this function.

File: tests/testsuite/future_incompat_report.rs [hunk 4]
```diff
@@ -126,21 +136,32 @@ frequency = 'never'
...
+[WARNING] the following packages contain code that will be rejected by a future version of Rust: foo v0.0.0 ([ROOT]/foo)
+...
+  - foo@0.0.0
+...
+")
             .run();
     }
 }
 
+#[allow(deprecated)]
```
Comment (eth3lbert) [inline tests/testsuite/future_incompat_report.rs:164]: `with_stderr_does_not_contain` is being used in this function

File: tests/testsuite/future_incompat_report.rs [hunk 5]
```diff
@@ -262,6 +314,7 @@ fn test_multi_crate() {
     assert_eq!(lines.next(), None);
 }
 
+#[allow(deprecated)]
```
Comment (eth3lbert) [inline tests/testsuite/future_incompat_report.rs:317]: `with_stdout_does_not_contain` is being used in this function.

File: tests/testsuite/fix.rs [hunk 6]
```diff
@@ -887,13 +901,7 @@ fn prepare_for_already_on_latest_stable() {
...
     p.cargo("fix --edition --allow-no-vcs")
-        .with_stderr_contains("[CHECKING] foo [..]")
-        .with_stderr_contains(&format!(
-            "\
-[WARNING] `src/lib.rs` is already on the latest edition ({latest_stable}), unable to migrate further
-",
-            latest_stable = latest_stable
-        ))
+        .with_stderr_data(str![])
         .run();
```
Comment (eth3lbert) [inline tests/testsuite/fix.rs:905]: I'm not sure why this isn't being overwritten. Has this functionality been tested at all?

Comment (weihanglo) [reply tests/testsuite/fix.rs:905]: No. I think we can add deprecated and leave off for now.

https://github.com/rust-lang/cargo/blob/c7cb45c0fdb65fc1d0252ecf4d7a8672667bc5a5/tests/testsuite/fix.rs#L881-L885

File: tests/testsuite/fix.rs [hunk 7]
```diff
@@ -529,19 +540,30 @@ fn fixes_missing_ampersand() {
...
-        .with_stderr_contains("[FIXED] src/main.rs (1 fix)")
-        .with_stderr_contains("[FIXED] examples/foo.rs (1 fix)")
-        .with_stderr_contains("[FIXED] tests/a.rs (1 fix)")
-        .with_stderr_contains("[FINISHED] [..]")
+        .with_stderr_data(str![[r#"
+...
+[FIXED] src/lib.rs ([..] fixes)
+...
+"#]])
+        .with_stderr_data(
```
[the remark this answers is not in the document]
Comment (eth3lbert) [reply tests/testsuite/fix.rs:554]: Oh, I forgot to mention this!  It sometimes generate two entries:
```
   1    1 | [COMPILING] foo v0.0.1 ([ROOT]/foo)
   2    2 | [FIXED] build.rs (1 fix)
   3    3 | [FIXED] src/main.rs (1 fix)
   4    4 | [FIXED] examples/foo.rs (1 fix)
   5    5 | [FIXED] tests/a.rs (1 fix)
   6    6 | [FIXED] src/lib.rs ([..]fix[..])
   7    7 | [FINISHED] `dev` profile [unoptimized + debuginfo] target(s) in [ELAPSED]s
        8 + [FIXED] src/lib.rs (1 fix)
```

Do you prefer something looks more like this?
``` rust
        .with_stderr_data(
            str![[r#"
[COMPILING] foo v0.0.1 ([ROOT]/foo)
[FIXED] build.rs (1 fix)
[FIXED] src/lib.rs ([..]fix[..])
[FIXED] src/main.rs (1 fix)
[FIXED] examples/foo.rs (1 fix)
[FIXED] tests/a.rs (1 fix)
[FINISHED] `dev` profile [unoptimized + debuginfo] target(s) in [ELAPSED]s
...
"#]]
```

**Comment by eth3lbert** [comment]
Feedbacks addressed! Thanks for the review!

**Review by weihanglo — APPROVED** [review]

**Comment by weihanglo** [comment]
Thanks!


@bors r+

**Pull request merged**